### DISCUSSION TITLE
Testing OpenVINO Noise Suppression with Audacity 3.4.2

### DISCUSSION THREAD

**Discussion by petersampsonaudacity** [intel/openvino-plugins-ai-audacity#78] [Show and tell]
I tested Noise Suppression on a commercially released CD - the same track I used in #4 
This is Michelle Holding and Bonz with Michelle singing in Lancashire dialect and Bonz accompanying her on banjo, so no "noise" to speak or really.

Here the result was disappointing.  I ended up with what sounded like a poor instrument removal - vocals with occasional percussive thumps from the banjo but no banjo tune.

_Now some may say that removing banjo is no bad thing_ ;-) - _but for the purposes of this test ..._

**Comment by petersampsonaudacity** [comment]
So then I tried on on Sgt Pepper track-1 and it took out almost everything (I used default settings):
![image](https://github.com/intel/openvino-plugins-ai-audacity/assets/50205984/90686643-7bfd-4bb9-859d-ad818cdf6604)

**Comment by RyanMetcalfeInt8** [comment]
@petersampsonaudacity -- Thank you! It's a bit unclear from the documentation (sorry about that), but this noise suppression model was designed to remove background noise from spoken audio (as opposed to removing generic 'noise' from, say, musical tracks).

So in this case, it will treat all non-vocals as 'noise' -- so I think your results make sense.

With that said, it would be really nice to support a more generic 'noise removal' effect, which is what I think you may have expected from this to begin with. I'll keep an eye out for emerging open source AI models that we could leverage for this (as this is what it comes down to). 

If anyone has spotted anything that we could leverage here, please point it out. 

Thanks!

**Comment by petersampsonaudacity** [comment]
@RyanMetcalfeInt8  It night be useful if the user could also (optionally) see a track with the noise that is taken out "suppressed".

I say this based on my experience with Brian Davies' excellent ClickRepair which I used extensively when I was transcribing my LPs where being able to see and hear to noise/clicks that would be removed enables one to fine-tune the parameter settings
![image](https://github.com/intel/openvino-plugins-ai-audacity/assets/50205984/c7c4f39b-cd65-4ab4-ad60-ea8fb42a911f)
So here you can choose to hear
a) the noise
b) the noisy input signal
c) the de-noised output signal
and flit between them as processing takes place - most useful.

**Comment by petersampsonaudacity** [comment]
> So in this case, it will treat all non-vocals as 'noise' -- so I think your results make sense.

But with Sgt Pepper it also removed 95% or more of the vocals

**Comment by RyanMetcalfeInt8** [comment]
> @RyanMetcalfeInt8 It night be useful if the user could also (optionally) see a track with the noise that is taken out "suppressed".
> 
> I say this based on my experience with Brian Davies' excellent ClickRepair which I used extensively when I was transcribing my LPs where being able to see and hear to noise/clicks that would be removed enables one to fine-tune the parameter settings ![image](https://private-user-images.githubusercontent.com/50205984/291627986-c7c4f39b-cd65-4ab4-ad60-ea8fb42a911f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5OTkwODEsIm5iZiI6MTcwMjk5ODc4MSwicGF0aCI6Ii81MDIwNTk4NC8yOTE2Mjc5ODYtYzdjNGYzOWItY2Q2NS00YWI0LWFkNjAtZWE4ZmI0MmE5MTFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDE1MTMwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxY2E2MzllOTJlYTRhNjc3NzRiMGE5OGFkYjg3NGZjNGIwYjIyMjRkYTdiNmE0N2VhN2QyMDMxODIwMmNlNWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lZSdHxK7TmV0uM97UDiHE97TgarlI4veU0RzPaEPrHU) So here you can choose to hear a) the noise b) the noisy input signal c) the de-noised output signal and flit between them as processing takes place - most useful.

Interesting - This would be definitely be something useful, I agree. Thanks for sharing.

**Comment by petersampsonaudacity** [comment]
> It's a bit unclear from the documentation (sorry about that), but this noise suppression model was designed to remove background noise from spoken audio (as opposed to removing generic 'noise' from, say, musical tracks).

@RyanMetcalfeInt8 

In which case you should probably talk to Paul Licameli who is one of the Muse Audacity developers (and was, like me, a member of the former Audacity Team before Muse took over the project).

Paul is interested in VoiceOver and runs a Facebook group for Audacity VoiceOver users:
https://www.facebook.com/groups/562223697221079

You can find Paul on the Muse Discord or on their GitHub

**Comment by RyanMetcalfeInt8** [comment]
> > So in this case, it will treat all non-vocals as 'noise' -- so I think your results make sense.
> 
> But with Sgt Pepper it also removed 95% or more of the vocals

Hmm, I'll experiment with it some more. It could be that the model was trained on primarily 'spoken' audio.. I'm not quite sure... I think in general, more cutting edge / sophisticated noise removal models would make a difference here.

**Comment by RyanMetcalfeInt8** [comment]
> > It's a bit unclear from the documentation (sorry about that), but this noise suppression model was designed to remove background noise from spoken audio (as opposed to removing generic 'noise' from, say, musical tracks).
> 
> @RyanMetcalfeInt8
> 
> In which case you should probably talk to Paul Licameli who is one of the Muse Audacity developers (and was, like me, a member of the former Audacity Team before Muse took over the project).
> 
> Paul is interested in VoiceOver and runs a Facebook group for Audacity VoiceOver users: https://www.facebook.com/groups/562223697221079
> 
> You can find Paul on the Muse Discord or on their GitHub

Cool, thanks! I've actually been working quite closely with Paul over the past few months. He has been extremely helpful in answering questions & actually pushing changes to Audacity repo itself to help support this.
